staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -146,28 +146,28 @@ type LeaderElector struct {
 }
 
 // Run starts the leader election loop
-func (le *LeaderElector) Run(stop <-chan struct{}) {
+func (le *LeaderElector) Run(ctx context.Context) {
```

Comment:
You could also create an overload like this:
`func (le *LeaderElector) Run() { le.Run(context.TODO()) }`

Then updating clients to use the new thing would be much more graceful.

Reply:
Go doesn't support overloading. We discussed adding WithContext style functions to maintain compatibility but the package is documented as having an alpha API. https://github.com/kubernetes/kubernetes/pull/57932#pullrequestreview-126079945

If people feel strongly we can do this, but I wouldn't fight for it.

Reply:
Oops - I am still learning quite a bit about golang.

I am not concerned with backward compatibility, but I thought the author's life would have been easier by temporarily preserving backward compatibility.